### PR TITLE
#170947633 Implement view all accommodations && book

### PR DESCRIPTION
**Pull request by T2Wil**
#### What does this PR do?
- Add Booking feature
- Add like and unlike feature

#### Description of Task to be completed?
- [x] Implement home page with a list of available accommodations
- [x] Implement `book accommodation` feature homepage.

#### How should this be manually tested?
- Clone the project

       git clone https://github.com/Stackup-Rwanda/knights-bn-frontend.git

- Move to the branch

       git checkout ft-book-accommodation-170947633

-  Install dependencies

       npm install

- Change the `baseURL` in the `src/reduxthunk/index` to `https://knights-bn-b-bg-fix-ima-tqmgsc.herokuapp.com/` to use it with the latest fixes made on backend.

-  start the server

       npm run dev-start 

-  log in the system

        email: maxime.kagorora@gmail.com
        password: Niyonkuru@1

#### What are the relevant pivotal tracker stories?
[User should be able to book accommodation](https://www.pivotaltracker.com/story/show/170947633)
### Homepage

image

### Book accommodation

image

image

### CODE REVIEW

**Review by Kagorora — APPROVED** [review]

**Review by Niyonsengaeric — APPROVED** [review]

File: .babelrc [hunk 1]
```diff
@@ -3,7 +3,7 @@
   "plugins": [
     "@babel/plugin-proposal-class-properties",
     "@babel/plugin-transform-regenerator",
-    "@babel/plugin-transform-runtime",
-    "transform-remove-console"
+    "@babel/plugin-transform-runtime"
+    // "transform-remove-console"
```
Comment (kagabof) [inline .babelrc]: Why do we have commented code?

File: src/views/Home.js [hunk 2]
```diff
@@ -1,13 +1,118 @@
...
+import { Spinner } from 'react-bootstrap';
+import CardComponent from '../components/Card';
+import '../assets/styles/containers/home.scss';
+import '../assets/styles/containers/loginPage.scss';
+import { getAccommodations } from '../redux/actions/actions';
+import { callApiThunk as thunk } from '../redux/thunk';
+import NavBar from '../components/Nav';
+
+class Home extends React.Component {
+  constructor(props) {
```
Comment (kagabof) [inline src/views/Home.js]: Please use es6.

**Review by kagabof — CHANGES_REQUESTED** [review]
Please validate your props in all components.

File: src/views/Home.js (hunk 2 above)
Comment (T2Wil) [reply src/views/Home.js]: Solved,

File: .babelrc (hunk 1 above)
Comment (T2Wil) [reply .babelrc]: solved

File: src/components/BookingCard.js [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+import Proptypes from 'prop-types';
+import Line from './line';
+import calculateDifference from '../util/calculateDateDifference';
+import { callApiThunk as thunk } from '../redux/thunk';
+import { bookAccommodation } from '../redux/actions/actions';
+import '../assets/styles/containers/loginPage.scss';
+import Alert from './Alert';
+import DateInput from './DateInput';
+
+const BookingCard = (props) => {
```
Comment (kagabof) [inline src/components/BookingCard.js]: Descructur in this component.
```
const BookingCard = ({  pricePerNight, ..., .., }) => {

}
```

File: src/components/Card.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+import { Card, Button } from 'react-bootstrap';
+import Proptypes from 'prop-types';
+import locationImage from '../assets/images/location.jpg';
+import likeImage from '../assets/images/like.png';
+import dislikeImage from '../assets/images/dislike.png';
+import starImage from '../assets/images/star_bold.png';
+import '../assets/styles/components/card.scss';
+import '../assets/styles/components/btn.scss';
+
+const CardComponent = (props) => {
```
Comment (kagabof) [inline src/components/Card.js]: Descructur in this component.

```
const CardComponent = ({  pricePerNight, ..., .., }) => {

}
```

File: src/views/BookAccommodation.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import Proptypes from 'prop-types';
+import Nav from '../components/Nav';
+import BookingCard from '../components/BookingCard';
+import CarouselComponent from '../components/CarouselComponent';
+import '../assets/styles/containers/bookAccommodation.scss';
+
+const BookAccommodation = (props) => {
```
Comment (kagabof) [inline src/views/BookAccommodation.js]: Descructur in this component.

```
const BookAccommodation = ({  pricePerNight, ..., .., }) => {

}
```

**Review by kagabof — CHANGES_REQUESTED** [review]

File: src/components/BookingCard.js (hunk 3 above)
Comment (T2Wil) [reply src/components/BookingCard.js]: Correction made!

File: src/components/Card.js (hunk 4 above)
Comment (T2Wil) [reply src/components/Card.js]: Correction made.

File: src/views/BookAccommodation.js (hunk 5 above)
Comment (T2Wil) [reply src/views/BookAccommodation.js]: correction made!

**Review by kagabof — APPROVED** [review]
Great job.

**Pull request merged**